File: p053.hs
{- 
 - Solution to Project Euler problem 53
 - By Nayuki Minase
 - 
 - http://nayuki.eigenstate.org/page/project-euler-solutions
 - https://github.com/nayuki/Project-Euler-solutions
 -}


factorial 0 = 1
factorial n = n * (factorial (n - 1))

binomial n k = div (factorial n) ((factorial (n - k)) * (factorial k))

ans = sum [1 | n <- [1..100], r <- [1..n], binomial n r > 1000000]

main = putStrLn (show ans)

File: p053.py
# 
# Solution to Project Euler problem 53
# by Project Nayuki
# 
# http://www.nayuki.io/page/project-euler-solutions
# https://github.com/nayuki/Project-Euler-solutions
# 

import eulerlib


def compute():
	ans = 0
	for n in range(1, 101):
		for k in range(0, n + 1):
			if binomial(n, k) > 1000000:
				ans += 1
	return str(ans)


def binomial(n, k):
	return eulerlib.factorial(n) // (eulerlib.factorial(k) * eulerlib.factorial(n - k))


if __name__ == "__main__":
	print(compute())
